### ISSUE TITLE
python3 error, version 0.5, need to add the decode logic for (byte <-> str)

### ISSUE THREAD

**Issue by MarkHe1222** [anarchy9388/scan#17]
Process Process-13:45:
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scan3.py", line 775, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''
Process Process-13:46:
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scan3.py", line 775, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''
Process Process-13:51:
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scan3.py", line 775, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''
Process Process-13:53:
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scan3.py", line 775, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''
Process Process-13:69:
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "scan3.py", line 775, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''

**Comment by anarchy9388** [comment]
```
Traceback (most recent call last):
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/lib64/python3.4/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "./scan3.py", line 319, in __parse_candidate_block
    write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]), 16)
ValueError: invalid literal for int() with base 16: b''
```

**Comment by anarchy9388** [comment]
@MarkHe1222 

If binascii receives an empty bytes string for example 
```
b''
```
Then it doesn't return a hex representation so the int() function fails.
Here its getting an empty byte from target byte because target byte is not long enough.

Added some logging:
```
lglf132:/tmp/mark # cat /tmp/scan_busy_blocks_2019-07-15-21-42-20.log | grep ERROR
2019-07-15 21:42:55,694 scan3.py[line:323] ERROR candidate_block: file: /dae/uuid-b8d496c8-e542-4d03-9860-f0109bf4dfb0/0045, position: 10737418236, size: 10240, partition: b8d496c8-e542-4d03-9860-f0109bf4dfb0, blockBin: 0045, end_position:10737418240,
2019-07-15 21:42:55,694 scan3.py[line:324] ERROR b''
2019-07-15 21:42:55,694 scan3.py[line:325] ERROR target_byte len: 4
2019-07-15 21:42:55,694 scan3.py[line:326] ERROR 4
2019-07-15 21:42:55,694 scan3.py[line:327] ERROR 12
2019-07-15 21:42:55,694 scan3.py[line:328] ERROR b''
2019-07-15 21:42:55,696 scan3.py[line:323] ERROR candidate_block: file: /dae/uuid-b8d496c8-e542-4d03-9860-f0109bf4dfb0/0045, position: 10737418237, size: 10240, partition: b8d496c8-e542-4d03-9860-f0109bf4dfb0, blockBin: 0045, end_position:10737418240,
2019-07-15 21:42:55,696 scan3.py[line:324] ERROR b''
2019-07-15 21:42:55,697 scan3.py[line:325] ERROR target_byte len: 3
2019-07-15 21:42:55,697 scan3.py[line:326] ERROR 4
2019-07-15 21:42:55,697 scan3.py[line:327] ERROR 12
2019-07-15 21:42:55,697 scan3.py[line:328] ERROR b''
2019-07-15 21:42:55,698 scan3.py[line:323] ERROR candidate_block: file: /dae/uuid-b8d496c8-e542-4d03-9860-f0109bf4dfb0/0045, position: 10737418238, size: 10240, partition: b8d496c8-e542-4d03-9860-f0109bf4dfb0, blockBin: 0045, end_position:10737418240,
2019-07-15 21:42:55,698 scan3.py[line:324] ERROR b''
2019-07-15 21:42:55,698 scan3.py[line:325] ERROR target_byte len: 2
2019-07-15 21:42:55,698 scan3.py[line:326] ERROR 4
2019-07-15 21:42:55,698 scan3.py[line:327] ERROR 12
2019-07-15 21:42:55,699 scan3.py[line:328] ERROR b''
2019-07-15 21:42:55,700 scan3.py[line:323] ERROR candidate_block: file: /dae/uuid-b8d496c8-e542-4d03-9860-f0109bf4dfb0/0045, position: 10737418239, size: 10240, partition: b8d496c8-e542-4d03-9860-f0109bf4dfb0, blockBin: 0045, end_position:10737418240,
2019-07-15 21:42:55,700 scan3.py[line:324] ERROR b''
2019-07-15 21:42:55,700 scan3.py[line:325] ERROR target_byte len: 1
2019-07-15 21:42:55,700 scan3.py[line:326] ERROR 4
2019-07-15 21:42:55,700 scan3.py[line:327] ERROR 12
2019-07-15 21:42:55,700 scan3.py[line:328] ERROR b''
lglf132:/tmp/mark #
```

And the code:

```

            try:
                stored_checksum_1 = binascii.hexlify(target_byte[write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum])
                stored_checksum = int(stored_checksum_1, 16)
            except:
                logging.error('candidate_block: file: {}, position: {}, size: {}, partition: {}, blockBin: {}, end_position:{}, '.format(
                               block_file, position, size, partition, blockBin, end_position))
                logging.error('{}'.format(stored_checksum_1))
                logging.error('target_byte len: {}'.format(len(target_byte)))
                logging.error('{}'.format(write_block_header + data_block_length))
                logging.error('{}'.format(write_block_header + data_block_length + write_block_checksum))
                logging.error('{}'.format(target_byte[write_block_header + data_block_length:write_block_header + data_block_length + write_block_checksum]))

                print(traceback.format_exc())
```

It just didn't do this in the last version because python3 is a bit different.

We can just except the ValueError and skip to next byte

**Comment by anarchy9388** [comment]
@MarkHe1222  Fixed in PR #18
